Assets/Scripts/Enemy/WaveSpawner.cs

```diff
@@ -0,0 +1,112 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using TMPro;
+
+
+public class WaveSpawner : MonoBehaviour
+{
+    private EnemiesFactory enemyFactory;
+    private float waveTimer;
+    private int currentWave;
+    private GameManager gameManager;
+    private GameObject player;
+    private TextMeshProUGUI timerTxt;
+    [SerializeField] private List<Enemy> enemiesType;
+    [SerializeField] private List<Wave> waves;
+
+    void Awake()
+    {
+        enemyFactory = new EnemiesFactory(enemiesType);
+        player = GameObject.FindGameObjectWithTag("Player");
+        timerTxt = GameObject.FindGameObjectWithTag("WaveTimer").GetComponent<TextMeshProUGUI>();
+        gameManager = gameObject.GetComponent<GameManager>();
+    }
+
+
+    void Start() 
+    {
+        currentWave = 0;
+        waveTimer = 0f;
+        CreateWaves();
+        enemyFactory.GenerateEnemies(waves[currentWave].NbEnemies);
+    }
+
+    void Update()
+    {
+        if(player.activeSelf)
+        {
+            if(!enemyFactory.IsEmpty())
+                SpawnEnemy();
+            else if(waveTimer <= 0)
+            {   
+                Nextwave();
+            }
+        }
+    }
+
+
+
+    void CreateWaves()
+    {
+        waves.Add(new Wave(5, 5));
+        waves.Add(new Wave(7, 5));
+        waves.Add(new Wave(8, 7));
+        waves.Add(new Wave(10, 7));
+        waves.Add(new Wave(10, 10));
+        waves.Add(new Wave(10, 7));
+    }
+
+    void Nextwave()
+    {
+        waveTimer = 0;
+        currentWave = (currentWave + 1) % waves.Count; 
+        enemyFactory.GenerateEnemies(waves[currentWave].NbEnemies);   
+        StartCoroutine(StartCountdown(waves[currentWave].WaveTime));
+    }
+
+
+    // void FixedUpdate()
+    // {
+    //     waveTimer += Time.deltaTime; 
+    // }
+
+    public IEnumerator StartCountdown(float countdownValue)
+    {
+        waveTimer = countdownValue;
+        while (waveTimer > 0)
+        {
+            Debug.Log("Countdown: " + waveTimer);
+            yield return new WaitForSeconds(1.0f);
+            waveTimer--;
+            DisplayTime(waveTimer);
+
+        }
+        
+    }
+
+    void DisplayTime(float time)
+    {
+        float minutes = Mathf.FloorToInt(time / 60);  
+        float seconds = Mathf.FloorToInt(time % 60);
```

Comment:
Propositon (à tester):

ton modulo fait derrière `time - ((time / 60) * 60)`, alors tu peux reprendre le calcul de minutes `(time / 60)`

```suggestion
        float seconds = time - (minutes * 60);
```

Reply:
en fait, avons nous besoin du calcul des minutes, le timer ne se rend jamais à une minute

Reply:
> en fait, avons nous besoin du calcul des minutes, le timer ne se rend jamais à une minute et voir [#25 (comment)](https://github.com/Etienne-Perinet/SherbrookeGameJam/pull/25#issuecomment-1104441127) concernant l'affichage

Ça sera à voir quand on aura fait plus de tests avec la vitesse des ennemis qui augmente.